cmd/configure_test.go

```diff
@@ -1,106 +1,82 @@
 package cmd
 
 import (
-	"fmt"
-	"io/ioutil"
-	"os"
 	"testing"
 
 	"github.com/exercism/cli/config"
-	"github.com/spf13/cobra"
 	"github.com/stretchr/testify/assert"
 )
 
 func TestConfigure(t *testing.T) {
-	// Make sure we put back the config env var.
-	cfgHomeKey := "EXERCISM_CONFIG_HOME"
-	cfgHome := os.Getenv(cfgHomeKey)
-	defer os.Setenv(cfgHomeKey, cfgHome)
-
-	// Make sure we put back the real command-line arguments.
-	osArgs := os.Args
-	defer func() {
-		os.Args = osArgs
-	}()
-
-	// Set up a bogus root command.
-	fakeCmd := &cobra.Command{}
-	// Add the real configureCmd to it.
-	fakeCmd.AddCommand(configureCmd)
-
 	tests := []struct {
+		desc           string
 		args           []string
 		existingUsrCfg *config.UserConfig
 		expectedUsrCfg *config.UserConfig
 		existingAPICfg *config.APIConfig
 		expectedAPICfg *config.APIConfig
 	}{
 		{
-			// It writes the flags when there is no config file.
+			desc:           "It writes the flags when there is no config file.",
 			args:           []string{"fakeapp", "configure", "--token", "a", "--workspace", "/a", "--api", "http://example.com"},
 			existingUsrCfg: nil,
 			expectedUsrCfg: &config.UserConfig{Token: "a", Workspace: "/a"},
 			existingAPICfg: nil,
 			expectedAPICfg: &config.APIConfig{BaseURL: "http://example.com"},
 		},
 		{
-			// It overwrites the flags in the config file.
+			desc:           "It overwrites the flags in the config file.",
 			args:           []string{"fakeapp", "configure", "--token", "b", "--workspace", "/b", "--api", "http://example.com/v2"},
 			existingUsrCfg: &config.UserConfig{Token: "token-b", Workspace: "/workspace-b"},
 			expectedUsrCfg: &config.UserConfig{Token: "b", Workspace: "/b"},
 			existingAPICfg: &config.APIConfig{BaseURL: "http://example.com/v1"},
 			expectedAPICfg: &config.APIConfig{BaseURL: "http://example.com/v2"},
 		},
 		{
-			args: []string{"fakeapp", "configure", "--token", "c"},
-			// It overwrites the flags that are passed without losing the ones that are not.
+			desc:           "It overwrites the flags that are passed, without losing the ones that are not.",
+			args:           []string{"fakeapp", "configure", "--token", "c"},
 			existingUsrCfg: &config.UserConfig{Token: "token-c", Workspace: "/workspace-c"},
 			expectedUsrCfg: &config.UserConfig{Token: "c", Workspace: "/workspace-c"},
-			// It gets the default API base URL.
+		},
+		{
+			desc:           "It gets the default API base URL.",
```

Comment:
I split apart the last test case into two separate ones. I realized that they were completely unrelated.